src/charge_controller/ChargeController.cpp

```diff
@@ -48,44 +48,83 @@ void ChargeController::process() {
 
     display.display();
 
-    // Display the recieved current and voltage from the charger.
-    //    uint16_t voltage = ((uint16_t) payload[1] << 8) | payload[0];
-    //    uint16_t current = ((uint16_t) payload[3] << 8) | payload[2];
-    //
-    //    log::LOGGER.log(log::Logger::LogLevel::DEBUG, "Payload: 0x%x, Current: %d. Voltage: %d", payload,  current, voltage);
-    //
-    //    display.setChargerCurrent(current);
-    //    display.setChargerVoltage(voltage);
-}
+    bms.update();
 
-uint8_t ChargeController::checkBMS() {
-    bool status = 0;
-
-    for (int i = 0; i < BMSManager::MAX_BMS_PACKS; i++) {
-        if (bms.isConnected(i)) {
-            if (!CHECK_IN_RANGE(bms.getBatteryVoltage(i), MAX_PACK_VOLTAGE, MIN_PACK_VOLTAGE)) {
-                status |= BAD_PACK_VOLTAGE;
-            }
-            if (!CHECK_IN_RANGE(bms.getMaxCellVoltage(i), MAX_CELL_VOLTAGE, MIN_CELL_VOLTAGE)) {
-                status |= BAD_MAX_CELL_VOLTAGE;
-            }
-            if (!CHECK_IN_RANGE(bms.getMinCellVoltage(i), MAX_CELL_VOLTAGE, MIN_CELL_VOLTAGE)) {
-                status |= BAD_MIN_CELL_VOLTAGE;
-            }
-            if (!CHECK_IN_RANGE(bms.getBatteryMaxTemp(i), MAX_TEMPERATURE, MIN_TEMPERATURE)) {
-                status |= BAD_MAX_TEMP;
-            }
-            if (!CHECK_IN_RANGE(bms.getBatteryMinTemp(i), MAX_TEMPERATURE, MIN_TEMPERATURE)) {
-                status |= BAD_MIN_TEMP;
-            }
-
-            if (bms.faultDetected(i)) {
-                log::LOGGER.log(log::Logger::LogLevel::DEBUG, "Fault State, Pack (%d)", i);
-                status |= BAD_FAULT_STATE;
-            }
+    if (startButton.debounce(500)) {
+        log::LOGGER.log(log::Logger::LogLevel::DEBUG, "Start Pressed");
+        if (isCharging()) {
+            stopCharging();
+        } else {
+            startCharging();
         }
     }
 
+    if (bms.numConnected() != oldCount) {
+        log::LOGGER.log(log::Logger::LogLevel::DEBUG, "%d batteries connected", bms.numConnected());
+        oldCount = bms.numConnected();
+    }
+
+    if (time::millis() - lastHeartBeat > HEARBEAT_INTERVAL) {
+        switch (statusLED.readPin()) {
+        case IO::GPIO::State::LOW:
+            statusLED.writePin(IO::GPIO::State::HIGH);
+            break;
+        case IO::GPIO::State::HIGH:
+            statusLED.writePin(IO::GPIO::State::LOW);
+            break;
+        }
+
+        // Need to send heartbeat can message to the charger
+        sendChargerMessage();
+        lastHeartBeat = time::millis();
+    }
+}
+
+uint8_t ChargeController::checkBMS() {
+    uint8_t status = 0;
+
+    /*
+    TODO: was faulting on bms.getMaxCellVoltage(i)
```

Comment:
Why is this all commented out?

Reply:
I talked to Jonas about this. Basically, all of the CAN stuff is broken, so I'd rather have it handled in a separate PR